sql/catalog.go

```diff
@@ -123,32 +133,65 @@ func (d *Databases) Add(db Database) {
 }
 
 // Table returns the Table with the given name if it exists.
-func (d Databases) Table(ctx context.Context, dbName string, tableName string, idxReg *IndexRegistry) (Table, error) {
+func (d Databases) Table(ctx *Context, dbName string, tableName string, idxReg *IndexRegistry) (Table, error) {
 	db, err := d.Database(dbName)
 	if err != nil {
 		return nil, err
 	}
 
-	tableName = strings.ToLower(tableName)
-
 	tbl, ok, err := db.GetTableInsensitive(ctx, tableName)
 
 	if err != nil {
 		return nil, err
 	} else if !ok {
-		tableNames, err := db.GetTableNames(ctx)
+		return nil, suggestSimilarTables(db, ctx, tableName)
+	}
 
-		if err != nil {
-			return nil, err
-		}
+	return tbl, nil
+}
 
-		similar := similartext.Find(tableNames, tableName)
-		return nil, ErrTableNotFound.New(tableName + similar)
+func suggestSimilarTables(db Database, ctx *Context, tableName string) error {
+	tableNames, err := db.GetTableNames(ctx)
+	if err != nil {
+		return err
+	}
+
+	similar := similartext.Find(tableNames, tableName)
+	return ErrTableNotFound.New(tableName + similar)
+}
+
+func (d Databases) TableAsOf(ctx *Context, dbName string, tableName string, time interface{}) (Table, error) {
```

Comment:
Comment on Exported items.

Reply:
Don't love using time as a variable name when time is the name of one of the go standard packages, means in the future if somebody imports that package they may have to rename it.

Reply:
Done.